tools/myst/host/strace.c

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "strace.h"
+#include <memory.h>
+#include "utils.h"
+
+#include <myst/syscall.h>
+
+int myst_parse_strace_config(
+    int* argc,
+    const char** argv,
+    myst_strace_config_t* strace_config)
+{
+    int ret = -1;
+    const char* filter = NULL;
+    char* filter_copy = NULL;
+
+    if (cli_getopt(argc, argv, "--strace-failing", NULL) == 0)
+    {
+        strace_config->trace_failing = 1;
+        strace_config->filter = 1;
+        ret = 0;
+    }
+
+    if (cli_getopt(argc, argv, "--strace-filter", &filter) == 0 && filter)
+    {
+        filter_copy = strdup(filter);
```

Comment:
Not a requirement for this PR but I sometimes find it easier to use ``myst_strsplit()`` from ``<myst/strings.h>``.

```
int myst_strsplit(
    const char* str,
    const char* delim,
    char*** toks,
    size_t* ntoks);
```

You can later pass ``toks`` to free.

Reply:
Nice idea. Done.